test/ipynb/mpl/circuit/test_circuit_matplotlib_drawer.py

```diff
@@ -621,6 +621,33 @@ def test_overwide_gates(self):
         circuit.initialize(initial_state)
         self.circuit_drawer(circuit, filename="wide_params.png")
 
+    def test_user_ax_subplot(self):
+        """Test for when user supplies ax for a subplot"""
+        import matplotlib.pyplot as plt
+
+        fig = plt.figure(1, figsize=(6, 4))
+        fig.patch.set_facecolor("white")
+        fig.add_subplot(1, 2, 1)
+        ax2 = fig.add_subplot(1, 2, 2)
+
+        circuit = QuantumCircuit(4)
+        circuit.h(0)
+        circuit.cx(0, 1)
+        circuit.h(1)
+        circuit.cx(1, 2)
+        plt.close(fig)
```

Comment:
Maybe a slightly more realistic scenario?
```suggestion
        fig = plt.figure(1, figsize=(6, 4))
        ax1 = fig.add_subplot(1, 2, 1)

        circuit = QuantumCircuit(2)
        circuit.h(0)
        circuit.cx(0, 1)
        circuit.draw('mpl', ax=ax1)

        ax2 = fig.add_subplot(1, 2, 2)

        circuit = QuantumCircuit(4)
        circuit.h(0)
        circuit.cx(0, 1)
        circuit.h(1)
        circuit.cx(1, 2)
        plt.close(fig)
```